examples/SpringTxFailoverExample/src/main/java/example/ExampleDaoImpl.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package example;
+
+import java.util.List;
+import java.util.Map;
+import javax.sql.DataSource;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+public class ExampleDaoImpl implements ExampleDao {
+  private DataSource dataSource;
```

Comment:
It might be better to make this class a Spring Component with the `@Component` annotation. Then make the `datasource` field `@Autowired` and get rid of the `setDataSource()` method. 

The `@Component` annotation on this class will make it so that you dont have to explicitly specify a bean for this class (and its interface).  So, the bean for `ExampleDao` in the `ExampleConfiguration` can be gotten rid of.